source/common/network/utility.cc

```diff
@@ -379,6 +379,7 @@ Address::InstanceConstSharedPtr Utility::getOriginalDst(Socket& sock) {
   }
 
   sockaddr_storage orig_addr;
+  memset(&orig_addr, 0, sizeof(orig_addr));
   socklen_t addr_len = sizeof(sockaddr_storage);
   int status;
 
```

Comment:
I noticed that the addr_len output argument below seems to be ignored, not passed on to addressFromSockAddr.  It's a bit strange that sock.getSocketOption does not set all relevant fields and results in read from uninitialized memory.

There seems to be other uses of sockaddr_storage in envoy code that don't do 0 initialization of that struct.

Reply:
I took a look at network/address_impl.cc and this does seem strange - it looks like addressFromSockAddr is supposed to check if ss_len is the correct size based on domain, but passing in ss_len = 0 guarantees that the corresponding RELEASE_ASSERTs never fail. Is that intended?

Reply:
Passing in 0 for ss_len seems suspicious.  It may be worth replacing said RELEASE_ASSERTs with more graceful error handling.

Reply:
I agree that doing something more intelligent, like graceful error handling, would make more sense.